main.go

```diff
@@ -61,6 +63,18 @@ func main() {
 		err = datasetsize.DatasetSize(args)
 	case "list":
 		err = list.List(args)
+	case "login":
+		deviceLogin, err := login.NewDeviceLogin(args)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to contact authentication service, reason: %v\n", err)
+		}
```

Comment:
If this fails there is no need to try to do the actual authentication request.

Reply:
This was also changed as I refactored to move that block out of main.go